sensing/image_diagnostics/include/image_diagnostics/image_diagnostics_node.hpp

```diff
@@ -0,0 +1,98 @@
+// Copyright 2022 TIER IV, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef IMAGE_DIAGNOSTICS__IMAGE_DIAGNOSTICS_NODE_HPP_
+#define IMAGE_DIAGNOSTICS__IMAGE_DIAGNOSTICS_NODE_HPP_
+
+#include <diagnostic_updater/diagnostic_updater.hpp>
+#include <image_transport/image_transport.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <sensor_msgs/msg/image.hpp>
+#include <tier4_debug_msgs/msg/float32_multi_array_stamped.hpp>
+#include <tier4_debug_msgs/msg/float32_stamped.hpp>
+#include <tier4_debug_msgs/msg/int32_stamped.hpp>
+
+#include <cv_bridge/cv_bridge.h>
+
+#include <string>
+#include <unordered_map>
+namespace image_diagnostics
+{
+using diagnostic_updater::DiagnosticStatusWrapper;
+using diagnostic_updater::Updater;
+
+enum Image_State : uint8_t { NORMAL = 0, DARK, BLOCKAGE, LOW_VIS, BACKLIGHT };
+std::unordered_map<std::string, cv::Scalar> state_color_map_ = {
+  {"NORMAL", cv::Scalar(100, 100, 100)},  {"DARK", cv::Scalar(0, 0, 0)},
+  {"BLOCKAGE", cv::Scalar(0, 0, 200)},    {"LOW_VIS", cv::Scalar(0, 200, 200)},
+  {"BACKLIGHT", cv::Scalar(200, 0, 200)}, {"BORDER", cv::Scalar(255, 255, 255)}};
+
+class ImageDiagNode : public rclcpp::Node
+{
+private:
+  void ImageChecker(const sensor_msgs::msg::Image::ConstSharedPtr input_image_msg);
+  void shiftImage(cv::Mat & img);
+  void onImageDiagChecker(DiagnosticStatusWrapper & stat);
+
+  int num_of_regions_normal = 0;
```

Comment:
```suggestion
  int num_of_regions_normal_ = 0;
```

Reply:
It may be better to use struct, since there are too many parameters

Reply:
thanks for comment, I will change to struct